src/graphics/uniform-buffer-format.js

```diff
@@ -102,17 +120,19 @@ class UniformBufferFormat {
 
         // TODO: optimize uniforms ordering
 
-        let byteSize = 0;
+        let offset = 0;
         for (let i = 0; i < uniforms.length; i++) {
             const uniform = uniforms[i];
-            uniform.offset = byteSize / 4;
-            byteSize += uniform.byteSize;
+            uniform.calculateOffset(offset);
+            offset = uniform.offset * 4 + uniform.byteSize;
```

Comment:
And then the multiplication by 4?

Reply:
in this instance, I need byte size, so I convert it. In UniformBuffer I don't multiply it when writing to the buffer.

Reply:
Is the uniform offset in indices and not in bytes?

Reply:
yep, as that's how I use it each frame .. so optimizing for that.